user/overriding-packages.rst

```diff
@@ -0,0 +1,282 @@
+If their build process stores some information about the non-leaf package, such as an expected ABI, then undefined behavior can happen at runtime.
+
+Problems caused by packages remembering information at build time can be avoided by overriding every package that directly or indirectly depends on the one you actually want to override.
+The group of overridden packages must span all underlays.
+
+Say there are 3 workspaces, **A**, **B** and **C**, where **C** overlays **B** and **B** overlays **A**.
+**A** contains packages ``foo`` and ``baz`` where ``baz`` depends on ``foo``.
+**B** contains packages ``ping`` that depends on ``foo`` and  ``pong`` that depends on ``baz``.
+**C** is the workspace being built.
+If you want to override ``foo`` then you should also override ``baz``, ``ping``, and ``pong``.
+
+Make sure overridden Python packages do not change entry point specifications
+*****************************************************************************
+
+Python packages may have `entry point specifications <https://packaging.python.org/en/latest/specifications/entry-points/>`_ in a ``setup.py`` or ``setup.cfg`` file.
+Make sure the package in the overlay has identical specifications to the version in the underlay, or only adds new ones.
+If any specification has been changed or removed then it may not be possible to override this package.
+
+How to make it easier for your users to override
+------------------------------------------------
+
+This section has advice for package authors about how to make easier for your users to use your package and override it or other pacakges.
+
+Install headers to a unique include directory
+*********************************************
+
+Install your package's headers to a unique folder rather than a shared folder.
+Say you're the author of a package ``foo``, and it has a header meant to be included like ``#include <foo/foo.hpp>``.
+Instead of installing the header to ``<prefix>/include/foo/foo.hpp``, install it to ``<prefix>/include/foo/foo/foo.hpp``.
+
+If you have a CMake package ``foo``  with the directory structure ``include/foo/foo.hpp``, then it can install its headers to a unique directory with this:
+
+.. code-block:: CMake
+
+  install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})
+
+All exported targets in your project need to export the unique include directory too.
+
+.. code-block:: CMake
+
+    target_include_directories(some_library_in_foo INTERFACE
+      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+      "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
+
+Dynamically link to libraries outside your package
+**************************************************
+
+If your package ``foo`` statically links to ``libbar.lib`` from package ``bar``, then users can't override ``bar`` without also overriding yours.
+Prefer dynamic linking to ``libbar.so`` instead.
```

Comment:
Unless I'm mistaken, this section is mixing Windows and Linux extensions, which is confusing. Either `.a`/`.so` or `.lib`/`.dll` would be clearer to me.

Reply:
Reworded to avoid platform specific extensions in 4cdaf32